api/CSVHandler.py

```diff
@@ -50,3 +48,44 @@ def test_split_train_test():
     train, test = split_train_test(data)
     assert np.shape(train) == (40, 4)
     assert np.shape(test) == (27, 4)
+
+def create_graph_nodes(data):
+    for line in data:
+        if (line[0].isdigit()) and ("->" not in line):
+            node = line.split("\"")[1].split("\\n")
+            node_id = int(line[0])
+            gini = ""
+            samples = ""
+            values = ""
+            expression = ""
+            for attr in node:
+                if 'gini' in attr:
+                    gini = float(attr.split(" ")[-1])
```

Comment:
I know we haven't run into it yet, but we could run into an issue where we end up with some odd trailing or leading whitespace. It might be useful to do `attr.strip().split(" ")[-1]` to ensure that this is taken care of.